### PR TITLE
Fix null pointer dereference in UpdateAddressInfo

### PR DESCRIPTION
**Pull request by dimitry-**
b/157410435

### CODE REVIEW

File: OrbitCore/SamplingProfiler.cpp [hunk 1]
```diff
@@ -426,7 +426,9 @@ void SamplingProfiler::UpdateAddressInfo(uint64_t address) {
   if (function != nullptr) {
     function_address = function->GetVirtualAddress();
     function_name = function->PrettyName();
-    address_info->function_name = function_name;
+    if (address_info != nullptr) {
```
Comment (ronaldfw) [inline OrbitCore/SamplingProfiler.cpp:429]: It doesn't change anything but for readability consider moving the update down to where the other updates happen (below line 444). Then you have a clear structure where here you are just filling function_address and function_name with the correct values and then you do all necessary updates to the data structures below. Will IMO be easier to maintain later.

Comment (dimitry-) [reply OrbitCore/SamplingProfiler.cpp]: done.

File: OrbitCore/SamplingProfiler.cpp [hunk 2]
```diff
@@ -436,6 +435,10 @@ void SamplingProfiler::UpdateAddressInfo(uint64_t address) {
     function_address = address;
   }
 
+  if (address_info != nullptr) {
```
Comment (ronaldfw) [inline OrbitCore/SamplingProfiler.cpp:438]: This now actually seems to be functionally different. Before you only updated address_info->function_name when function was not a nullptr. I don't know what is correct.

Comment (dimitry-) [reply OrbitCore/SamplingProfiler.cpp:438]: the only difference is that empty names are going to be updated with "???". I think this is fine.

**Pull request merged**